### ISSUE TITLE
[JENKINS-56773] High master CPU usage introduced by Blue Ocean 1.14.0

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/blueocean-plugin#4954]
<p>We went from BO 1.13.2 to BO 1.14.0.<br/>
After the upgrade, we have seen a major CPU increase on our Jenkins master, leading to really slow UI and job execution, with threads being locked - see screenshot. <br/>
<span class="image-wrap" style=""><a id="46572_thumb" href="https://issues.jenkins.io/secure/attachment/46572/46572_image-2019-03-27-10-31-38-911.png" title="image-2019-03-27-10-31-38-911.png" file-preview-type="image" file-preview-id="46572" file-preview-title="image-2019-03-27-10-31-38-911.png"><img src="https://issues.jenkins.io/secure/thumbnail/46572/_thumb_46572.png" style="border: 0px solid black" role="presentation"/></a></span> </p>

<p>Our master is configured to not run any job ("only build jobs with label expression matching this node").<br/>
Using Jenkins monitor, we saw that ~90% of the CPU time was consumed by calls to the rest endpoint <tt>/blue/rest/organizations/jenkins/computers</tt> - see screenshot.<br/>
 <span class="image-wrap" style=""><a id="46571_thumb" href="https://issues.jenkins.io/secure/attachment/46571/46571_image-2019-03-27-10-35-23-276.png" title="image-2019-03-27-10-35-23-276.png" file-preview-type="image" file-preview-id="46571" file-preview-title="image-2019-03-27-10-35-23-276.png"><img src="https://issues.jenkins.io/secure/thumbnail/46571/_thumb_46571.png" style="border: 0px solid black" role="presentation"/></a></span> </p>

<p>Looking at the release notes from 1.14.0, it seems that <a href="https://issues.jenkins.io/browse/JENKINS-52825" title="Cannot see executors in Blue Ocean UI" class="issue-link" data-issue-key="JENKINS-52825"><del>JENKINS-52825</del></a> has introduced that endpoint and should be an area to investigate.</p>

<p>We had to rollback BO to 1.13.2 which does not exhibit that behavior.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ojacques.png" title="ojacques's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ojacques">ojacques</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773" target="_blank">High master CPU usage introduced by Blue Ocean 1.14.0</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/olamy.jpg" title="olamy's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olamy">olamy</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: blueocean-plugin
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2020-10-27T06:59:00+00:00
<li><b>votes</b>: 7
<li><b>watchers</b>: 16
<li><b>imported</b>: 2025-12-01
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>We went from BO 1.13.2 to BO 1.14.0.
After the upgrade, we have seen a major CPU increase on our Jenkins master, leading to really slow UI and job execution, with threads being locked - see screenshot. 
<span class="image-wrap" style=""><a id="46572_thumb" href="https://issues.jenkins.io/secure/attachment/46572/46572_image-2019-03-27-10-31-38-911.png" title="image-2019-03-27-10-31-38-911.png" file-preview-type="image" file-preview-id="46572" file-preview-title="image-2019-03-27-10-31-38-911.png"><img src="https://issues.jenkins.io/secure/thumbnail/46572/_thumb_46572.png" style="border: 0px solid black" role="presentation"/></a></span> </p>

<p>Our master is configured to not run any job ("only build jobs with label expression matching this node").
Using Jenkins monitor, we saw that ~90% of the CPU time was consumed by calls to the rest endpoint <tt>/blue/rest/organizations/jenkins/computers</tt> - see screenshot.
 <span class="image-wrap" style=""><a id="46571_thumb" href="https://issues.jenkins.io/secure/attachment/46571/46571_image-2019-03-27-10-35-23-276.png" title="image-2019-03-27-10-35-23-276.png" file-preview-type="image" file-preview-id="46571" file-preview-title="image-2019-03-27-10-35-23-276.png"><img src="https://issues.jenkins.io/secure/thumbnail/46571/_thumb_46571.png" style="border: 0px solid black" role="presentation"/></a></span> </p>

<p>Looking at the release notes from 1.14.0, it seems that <a href="https://issues.jenkins.io/browse/JENKINS-52825" title="Cannot see executors in Blue Ocean UI" class="issue-link" data-issue-key="JENKINS-52825"><del>JENKINS-52825</del></a> has introduced that endpoint and should be an area to investigate.</p>

<p>We had to rollback BO to 1.13.2 which does not exhibit that behavior.</p></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
BlueOcean 1.14.0<br/>
Jenkins 1.168
```
</details>
<details><summary><i>4 attachments</i></summary>

- [computers_GET.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/47921/computers_GET.png)
  > ![computers_GET.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/47921/computers_GET.png)
- [image-2019-03-27-10-31-38-911.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/46572/image-2019-03-27-10-31-38-911.png)
  > ![image-2019-03-27-10-31-38-911.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/46572/image-2019-03-27-10-31-38-911.png)
- [image-2019-03-27-10-35-23-276.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/46571/image-2019-03-27-10-35-23-276.png)
  > ![image-2019-03-27-10-35-23-276.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/46571/image-2019-03-27-10-35-23-276.png)
- [screencapture-testing-a-markwaite-net-8080-manage-2020-10-27-13_02_47-edit.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/53077/screencapture-testing-a-markwaite-net-8080-manage-2020-10-27-13_02_47-edit.png)
  > ![screencapture-testing-a-markwaite-net-8080-manage-2020-10-27-13_02_47-edit.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/53077/screencapture-testing-a-markwaite-net-8080-manage-2020-10-27-13_02_47-edit.png)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ojacques.png" title="ojacques's avatar" /> ojacques:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=363633&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-363633">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>/cc <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rodrigc" class="user-hover" rel="rodrigc">rodrigc</a> - in case this can indeed be caused by <a href="https://issues.jenkins.io/browse/JENKINS-52825" title="Cannot see executors in Blue Ocean UI" class="issue-link" data-issue-key="JENKINS-52825"><del>JENKINS-52825</del></a>.</p>
</pre>
</details>

</ul>
</details>

<p>/cc <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rodrigc" class="user-hover" rel="rodrigc">rodrigc</a> - in case this can indeed be caused by <a href="https://issues.jenkins.io/browse/JENKINS-52825" title="Cannot see executors in Blue Ocean UI" class="issue-link" data-issue-key="JENKINS-52825"><del>JENKINS-52825</del></a>.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/halkeye.png" title="halkeye's avatar" /> halkeye:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=364039&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-364039">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>So had a bit of a chat with keith (original executor plugin author) and it was never really designed for more than a proof of concept on how to write a BlueOcean plugin.</p>

<p>We are pretty sure its caused by <a href="https://github.com/jenkinsci/blueocean-plugin/blob/e2dd04dcb3015063a2e31c8d37c90ee5c1ef4d98/blueocean-executor-info/src/main/js/ExecutorInfoService.js#L14" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/blob/e2dd04dcb3015063a2e31c8d37c90ee5c1ef4d98/blueocean-executor-info/src/main/js/ExecutorInfoService.js#L14</a> (which will grab executor info every time any pipeline does work)</p>

<p>So Plans:</p>
<ul>
	<li>I think the plan is going to be to drop making the executor plugin as a requirement, it'll have to be manually installed again (Sorry Craig)</li>
	<li>Put a throttle on the lookup so it doesn't call nearly as often.</li>
</ul>


<p>Since there is a work around in place until the next release, and I personally have a few pressing issues to deal with first, i'm going to leave it unassigned, but we'll get it fixed before we do the next release.</p>
</pre>
</details>

</ul>
</details>

<p>So had a bit of a chat with keith (original executor plugin author) and it was never really designed for more than a proof of concept on how to write a BlueOcean plugin.</p>

<p>We are pretty sure its caused by <a href="https://github.com/jenkinsci/blueocean-plugin/blob/e2dd04dcb3015063a2e31c8d37c90ee5c1ef4d98/blueocean-executor-info/src/main/js/ExecutorInfoService.js#L14" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/blob/e2dd04dcb3015063a2e31c8d37c90ee5c1ef4d98/blueocean-executor-info/src/main/js/ExecutorInfoService.js#L14</a> (which will grab executor info every time any pipeline does work)</p>

<p>So Plans:</p>
<ul>
	<li>I think the plan is going to be to drop making the executor plugin as a requirement, it'll have to be manually installed again (Sorry Craig)</li>
	<li>Put a throttle on the lookup so it doesn't call nearly as often.</li>
</ul>


<p>Since there is a work around in place until the next release, and I personally have a few pressing issues to deal with first, i'm going to leave it unassigned, but we'll get it fixed before we do the next release.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ojacques.png" title="ojacques's avatar" /> ojacques:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=364072&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-364072">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=halkeye" class="user-hover" rel="halkeye">halkeye</a>.</p>
</pre>
</details>

</ul>
</details>

<p>Thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=halkeye" class="user-hover" rel="halkeye">halkeye</a>.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/halkeye.png" title="halkeye's avatar" /> halkeye:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=364467&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-364467">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://github.com/jenkinsci/blueocean-plugin/pull/1947" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/1947</a></p>
</pre>
</details>

</ul>
</details>

<p><a href="https://github.com/jenkinsci/blueocean-plugin/pull/1947" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/1947</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>dgoetsch:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=364917&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-364917">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This impacted our company and we had to do the same rollback.</p>

<p>we went from 1.6.x to 1.14.0, the next work day jenkins became unresponsive under load, and our jvm monitoring told us that the blue ocean rest computers endpoint was using most of the cpu time.</p>

<p>We rolled back to 1.13.2 and disabled the Blue Ocean Executor Info plugin and our Jenkins master is far more responsive and page loads for blue ocean are much lower.</p>

<p>Our builds have 3 stages, the first one is a single stage, the second one has 3 parallel stages, and the third one has 50 parallel stages, 32 of which are computed dynamically and launched with the "node" step.</p>

<p>Please revert the change to make the Blue Ocean Executor Info plugin a required dependency.  I'm sure it would be okay on a smaller system, but for a system like ours it is untenable.</p>
</pre>
</details>

</ul>
</details>

<p>This impacted our company and we had to do the same rollback.</p>

<p>we went from 1.6.x to 1.14.0, the next work day jenkins became unresponsive under load, and our jvm monitoring told us that the blue ocean rest computers endpoint was using most of the cpu time.</p>

<p>We rolled back to 1.13.2 and disabled the Blue Ocean Executor Info plugin and our Jenkins master is far more responsive and page loads for blue ocean are much lower.</p>

<p>Our builds have 3 stages, the first one is a single stage, the second one has 3 parallel stages, and the third one has 50 parallel stages, 32 of which are computed dynamically and launched with the "node" step.</p>

<p>Please revert the change to make the Blue Ocean Executor Info plugin a required dependency.  I'm sure it would be okay on a smaller system, but for a system like ours it is untenable.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>sorenfriis:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=370428&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-370428">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>We are still seeing a very high amount of CPU time spend servicing the following endpoint on Blue Ocean 1.17.0</p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>/blue/rest/organizations/jenkins/computers/
</pre>
</div></div>
<p><span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/47921/47921_computers_GET.png" title="computers_GET.png attached to JENKINS-56773">computers_GET.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></p>

<p>I tried blocking the endpoint in my browser, and everything seemed to function normally on Blue Ocean pages.
Can the polling of this endpoint be disabled or reduced?</p>
</pre>
</details>

</ul>
</details>

<p>We are still seeing a very high amount of CPU time spend servicing the following endpoint on Blue Ocean 1.17.0</p>


```
/blue/rest/organizations/jenkins/computers/

```
<p><span class="nobr"><a href="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/47921/computers_GET.png" title="computers_GET.png attached to JENKINS-56773">computers_GET.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></p>

<p>I tried blocking the endpoint in my browser, and everything seemed to function normally on Blue Ocean pages.<br/>
Can the polling of this endpoint be disabled or reduced?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/halkeye.png" title="halkeye's avatar" /> halkeye:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=370462&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-370462">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>disable the executor plugin, its no longer required</p>
</pre>
</details>

</ul>
</details>

<p>disable the executor plugin, its no longer required</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=377257&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-377257">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Removing a plugin from the aggregator/meta plugin doesn't remove the broken functionality from any instances that have it installed.</p>

<p>Like ci.jenkins.io which got shit for the last several weeks for its recent stability problems. Imagine my surprise seeing dozens of threads hanging in the endpoint mentioned in this issue, and how this issue has been resolved.</p>

<p>At the very least, admins that generally take care of their instance need to be informed about this with an admin monitor.</p>
</pre>
</details>

</ul>
</details>

<p>Removing a plugin from the aggregator/meta plugin doesn't remove the broken functionality from any instances that have it installed.</p>

<p>Like ci.jenkins.io which got shit for the last several weeks for its recent stability problems. Imagine my surprise seeing dozens of threads hanging in the endpoint mentioned in this issue, and how this issue has been resolved.</p>

<p>At the very least, admins that generally take care of their instance need to be informed about this with an admin monitor.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=383832&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-383832">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>As far as I can tell, <a href="https://github.com/jenkinsci/blueocean-plugin/releases/tag/blueocean-parent-1.21.0" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/releases/tag/blueocean-parent-1.21.0</a> includes this change.  Thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick" class="user-hover" rel="jglick">jglick</a> for that pull request!  I think this can be marked 'Closed' in release blue ocean 1.21.0 with a code fix to prevent the issue.</p>

<p>Can one of the users that encountered this problem confirm that it is now resolved?</p>
</pre>
</details>

</ul>
</details>

<p>As far as I can tell, <a href="https://github.com/jenkinsci/blueocean-plugin/releases/tag/blueocean-parent-1.21.0" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/releases/tag/blueocean-parent-1.21.0</a> includes this change.  Thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick" class="user-hover" rel="jglick">jglick</a> for that pull request!  I think this can be marked 'Closed' in release blue ocean 1.21.0 with a code fix to prevent the issue.</p>

<p>Can one of the users that encountered this problem confirm that it is now resolved?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>mwos:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=387692&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-387692">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>We use Blue Ocean and Blue Ocean Executor Info plugins in version 1.22.0 and still have a problem with </p>

<p>```</p>

<p>/blue/rest/organizations/jenkins/computers/</p>

<p>```</p>

<p>endpoint. It is called very often and hangs for ~5sec each time. Not sure why informations about all executors is needed in the first place when opened blue ocean for a given job and a run. Those informations do not seem to be utilised anywhere in the view.</p>
</pre>
</details>

</ul>
</details>

<p>We use Blue Ocean and Blue Ocean Executor Info plugins in version 1.22.0 and still have a problem with </p>

<p>```</p>

<p>/blue/rest/organizations/jenkins/computers/</p>

<p>```</p>

<p>endpoint. It is called very often and hangs for ~5sec each time. Not sure why informations about all executors is needed in the first place when opened blue ocean for a given job and a run. Those informations do not seem to be utilised anywhere in the view.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=387693&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-387693">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mwos" class="user-hover" rel="mwos">mwos</a> sorry that you've encountered the problem.  Per the <a href="https://issues.jenkins-ci.org/browse/JENKINS-56773?focusedCommentId=370462&amp;page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-370462" class="external-link" target="_blank" rel="nofollow noopener">comment from Gavin Mogan</a>, the e Blue Ocean Executor Info plugin is no longer required.  You might be able to resolve the issue on your installation by removing the Blue Ocean Executor Info plugin.  It is not installed on ci.jenkins.io (likely because Daniel Beck removed it).  I was able to remove it from my Jenkins instance without seeing any harmful side effects.</p>

<p>I agree with Daniel that it would be much better to have an administrative monitor or a Jenkins Health Advisor warning that Blue Ocean Executor Info is not required and may cause performance issues.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mwos" class="user-hover" rel="mwos">mwos</a> sorry that you've encountered the problem.  Per the <a href="https://issues.jenkins-ci.org/browse/JENKINS-56773?focusedCommentId=370462&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-370462" class="external-link" target="_blank" rel="nofollow noopener">comment from Gavin Mogan</a>, the e Blue Ocean Executor Info plugin is no longer required.  You might be able to resolve the issue on your installation by removing the Blue Ocean Executor Info plugin.  It is not installed on ci.jenkins.io (likely because Daniel Beck removed it).  I was able to remove it from my Jenkins instance without seeing any harmful side effects.</p>

<p>I agree with Daniel that it would be much better to have an administrative monitor or a Jenkins Health Advisor warning that Blue Ocean Executor Info is not required and may cause performance issues.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>baychris:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=391610&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-391610">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Same issue. Enabling the executor plugin made our Jenkins instance unusable for a couple of days until we tracked the issue down to this bug and disabled it. Our Jenkins master took a massive performance hit with this plugin enabled. Please remove it if it's not required. </p>
</pre>
</details>

</ul>
</details>

<p>Same issue. Enabling the executor plugin made our Jenkins instance unusable for a couple of days until we tracked the issue down to this bug and disabled it. Our Jenkins master took a massive performance hit with this plugin enabled. Please remove it if it's not required. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/halkeye.png" title="halkeye's avatar" /> halkeye:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=391611&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-391611">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>remove it from what?</p>
</pre>
</details>

</ul>
</details>

<p>remove it from what?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/damien_coraboeuf.jpg" title="damien_coraboeuf's avatar" /> damien_coraboeuf:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=391633&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-391633">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Remove the plugin from your Jenkins installation.</p>
</pre>
</details>

</ul>
</details>

<p>Remove the plugin from your Jenkins installation.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=391638&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-391638">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>It's no longer a dependency in recent BO releases. You can just click the buttons.</p>

<p>I don't think the Jenkins project should be able to decide what gets uninstalled on people's instances.</p>
</pre>
</details>

</ul>
</details>

<p>It's no longer a dependency in recent BO releases. You can just click the buttons.</p>

<p>I don't think the Jenkins project should be able to decide what gets uninstalled on people's instances.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/eplodn1.jpg" title="eplodn1's avatar" /> eplodn1:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=398138&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-398138">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>We've been hit by this issue too, with the (unwelcome) addition of it causing any endpoint (browser) to waste CPU cycles and eventually get stuck in any installation having over ~30 nodes. We have hundreds of nodes, so any call to `/computers` endpoint is bringing 4-5 MB each, and there are several requests <em>each couple of seconds</em> from each computer having any BlueOcean interface open (of which we have above 150).</p>

<p>While I agree that Jenkins should not voluntarily remove plugins from people who installed them, this one deserves a separate treatment in my opinion, as a) it does not seem to bring any functionality to Jenkins by the virtue of it being a "demo" of how to write a BlueOcean plugin; and b) I believe most installations of it come from the people just selecting everything with "blueocean" in its name.</p>
</pre>
</details>

</ul>
</details>

<p>We've been hit by this issue too, with the (unwelcome) addition of it causing any endpoint (browser) to waste CPU cycles and eventually get stuck in any installation having over ~30 nodes. We have hundreds of nodes, so any call to `/computers` endpoint is bringing 4-5 MB each, and there are several requests <em>each couple of seconds</em> from each computer having any BlueOcean interface open (of which we have above 150).</p>

<p>While I agree that Jenkins should not voluntarily remove plugins from people who installed them, this one deserves a separate treatment in my opinion, as a) it does not seem to bring any functionality to Jenkins by the virtue of it being a "demo" of how to write a BlueOcean plugin; and b) I believe most installations of it come from the people just selecting everything with "blueocean" in its name.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=398139&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-398139">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<blockquote><p>this one deserves a separate treatment in my opinion</p></blockquote>
<p>I would rather not add an elaborate and certain to be controversial feature to Jenkins (core) for it to be used only once for an issue that was resolved a year ago, and then never again.</p>

<p>It would be plenty good enough for new Blue Ocean releases to show a warning to administrators if this plugin is still installed (and enabled), but that's up to its maintainers.</p>

<p> </p>
</pre>
</details>

</ul>
</details>

<blockquote><p>this one deserves a separate treatment in my opinion</p></blockquote>
<p>I would rather not add an elaborate and certain to be controversial feature to Jenkins (core) for it to be used only once for an issue that was resolved a year ago, and then never again.</p>

<p>It would be plenty good enough for new Blue Ocean releases to show a warning to administrators if this plugin is still installed (and enabled), but that's up to its maintainers.</p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> jglick:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=398156&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-398156">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Should suffice for BO maintainers (<a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=halkeye" class="user-hover" rel="halkeye">halkeye</a>? <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olamy" class="user-hover" rel="olamy">olamy</a>?) to just cut a “tombstone” release of this plugin with zero code and prominently marked as deprecated. Then people who simply update all plugins routinely would get the tombstone and no longer encounter issues; they could uninstall the plugin at some point for purposes of cleanup.</p>
</pre>
</details>

</ul>
</details>

<p>Should suffice for BO maintainers (<a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=halkeye" class="user-hover" rel="halkeye">halkeye</a>? <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olamy" class="user-hover" rel="olamy">olamy</a>?) to just cut a “tombstone” release of this plugin with zero code and prominently marked as deprecated. Then people who simply update all plugins routinely would get the tombstone and no longer encounter issues; they could uninstall the plugin at some point for purposes of cleanup.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=398192&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-398192">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<blockquote><p>marked as deprecated</p></blockquote>
<p>Good point, Jenkins 2.248 and newer also prominently shows which plugins are labeled deprecated; we could add this one: <a href="https://github.com/jenkins-infra/update-center2/pull/451" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkins-infra/update-center2/pull/451</a></p>

<p>(TBH it seems crazy to me that it's still getting new releases because it's still in the multimodule project…)</p>

<p> </p>
</pre>
</details>

</ul>
</details>

<blockquote><p>marked as deprecated</p></blockquote>
<p>Good point, Jenkins 2.248 and newer also prominently shows which plugins are labeled deprecated; we could add this one: <a href="https://github.com/jenkins-infra/update-center2/pull/451" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkins-infra/update-center2/pull/451</a></p>

<p>(TBH it seems crazy to me that it's still getting new releases because it's still in the multimodule project…)</p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>ethorsa:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=400048&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-400048">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I can't remove the plugin since it's still shown as a dependency of Blue Ocean. The enable and uninstall control are both greyed out. The Plugin is marked as deprecated though.</p>

<p>Jenkins 2.249.2, Blue Ocean / Blue Ocean Executor Info v1.24.2.</p>
</pre>
</details>

</ul>
</details>

<p>I can't remove the plugin since it's still shown as a dependency of Blue Ocean. The enable and uninstall control are both greyed out. The Plugin is marked as deprecated though.</p>

<p>Jenkins 2.249.2, Blue Ocean / Blue Ocean Executor Info v1.24.2.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/halkeye.png" title="halkeye's avatar" /> halkeye:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=400050&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-400050">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Please open a new issue if this is happening again, reviving old bugs is not going to get things addressed.</p>
</pre>
</details>

</ul>
</details>

<p>Please open a new issue if this is happening again, reviving old bugs is not going to get things addressed.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/olamy.jpg" title="olamy's avatar" /> olamy:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=400051&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-400051">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The plugin does NOTHING now and will be removed with next release.</p>
</pre>
</details>

</ul>
</details>

<p>The plugin does NOTHING now and will be removed with next release.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=400083&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-400083">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I'm seeing the same condition that is described by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ethorsa" class="user-hover" rel="ethorsa">ethorsa</a>.  Prior to Blue Ocean 1.24.1, I was able to uninstall the blueocean executor info plugin so that I could avoid this issue.  With the release of blueocean I'm unable to uninstall the blueocean executor info plugin.</p>

<p>I'm not seeing any issue from the blueocean-executor-info plugin itself, since (as noted by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olamy" class="user-hover" rel="olamy">olamy</a>) it now does nothing. It is an empty plugin.</p>

<p>I assume the next release of Blue Ocean (possibly 1.24.3) will remove the dependency on blueocean executor info plugin so that I can uninstall it.  Until then, the admin monitor warning will continue to be visible that tells me I have a deprecated plugin in my Jenkins installation.  I prefer to have no admin monitors raising alerts on my Jenkins installation, but there does not appear to be any way to resolve this one other than waiting for the next release of Blue Ocean.</p>

<p> <span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/53077/53077_screencapture-testing-a-markwaite-net-8080-manage-2020-10-27-13_02_47-edit.png" style="border: 0px solid black" /></span> </p>
</pre>
</details>

</ul>
</details>

<p>I'm seeing the same condition that is described by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ethorsa" class="user-hover" rel="ethorsa">ethorsa</a>.  Prior to Blue Ocean 1.24.1, I was able to uninstall the blueocean executor info plugin so that I could avoid this issue.  With the release of blueocean I'm unable to uninstall the blueocean executor info plugin.</p>

<p>I'm not seeing any issue from the blueocean-executor-info plugin itself, since (as noted by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olamy" class="user-hover" rel="olamy">olamy</a>) it now does nothing. It is an empty plugin.</p>

<p>I assume the next release of Blue Ocean (possibly 1.24.3) will remove the dependency on blueocean executor info plugin so that I can uninstall it.  Until then, the admin monitor warning will continue to be visible that tells me I have a deprecated plugin in my Jenkins installation.  I prefer to have no admin monitors raising alerts on my Jenkins installation, but there does not appear to be any way to resolve this one other than waiting for the next release of Blue Ocean.</p>

<p> <span class="image-wrap" style=""><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/53077/screencapture-testing-a-markwaite-net-8080-manage-2020-10-27-13_02_47-edit.png" style="border: 0px solid black" /></span> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/olamy.jpg" title="olamy's avatar" /> olamy:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=400089&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-400089">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I agree it's annoying... I will cut a new release next week so people will have enough time to notice this plugin has been deprecated <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</ul>
</details>

<p>I agree it's annoying... I will cut a new release next week so people will have enough time to notice this plugin has been deprecated <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>paulsbruce:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=400214&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-400214">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Also getting this in the past few days. To <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olamy" class="user-hover" rel="olamy">olamy</a> 's point, would like guidance on what to do here.</p>
</pre>
</details>

</ul>
</details>

<p>Also getting this in the past few days. To <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olamy" class="user-hover" rel="olamy">olamy</a> 's point, would like guidance on what to do here.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-56773?focusedId=400217&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-400217">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=paulsbruce" class="user-hover" rel="paulsbruce">paulsbruce</a> if you've upgraded to blue ocean 1.24.2 and restarted your Jenkins server since the upgrade, then the blueocean-executor-info plugin described in this bug report cannot be causing the high CPU usage.  With release 1.24.2, the blueocean-executor-info plugin contains no code.</p>

<p>The contents of the plugin file (hpi / jpi file) are:</p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>M Filemode      Length  Date         Time      File
- ----------  --------  -----------  --------  --------------------------------------------------------------------------
  drwxr-xr-x         0  27-Oct-2020  08:59:58  META-INF/
  -rw-r--r--       855  27-Oct-2020  08:59:56  META-INF/MANIFEST.MF
  drwxr-xr-x         0  27-Oct-2020  08:59:58  WEB-INF/
  drwxr-xr-x         0  27-Oct-2020  08:59:58  WEB-INF/lib/
  -rw-r--r--       531  27-Oct-2020  08:59:58  WEB-INF/licenses.xml
  -rw-r--r--      3648  27-Oct-2020  08:59:58  WEB-INF/lib/blueocean-executor-info.jar
  drwxr-xr-x         0  27-Oct-2020  08:59:58  META-INF/maven/
  drwxr-xr-x         0  27-Oct-2020  08:59:58  META-INF/maven/io.jenkins.blueocean/
  drwxr-xr-x         0  27-Oct-2020  08:59:58  META-INF/maven/io.jenkins.blueocean/blueocean-executor-info/
  -rw-r--r--      4010  27-Oct-2020  08:59:56  META-INF/maven/io.jenkins.blueocean/blueocean-executor-info/pom.xml
  -rw-r--r--       130  27-Oct-2020  08:59:58  META-INF/maven/io.jenkins.blueocean/blueocean-executor-info/pom.properties
- ----------  --------  -----------  --------  --------------------------------------------------------------------------
                  9174                         11 files
</pre>
</div></div>

<p>The contents of the jar file packaged in that plugin are:</p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>M Filemode      Length  Date         Time      File
- ----------  --------  -----------  --------  --------------------------------------------------------------------------
  drwxr-xr-x         0  27-Oct-2020  08:59:58  META-INF/
  -rw-r--r--       855  27-Oct-2020  08:59:56  META-INF/MANIFEST.MF
  -rw-r--r--        93  27-Oct-2020  08:59:56  jenkins-js-extension.json
  -rw-r--r--      1290  27-Oct-2020  08:36:28  package.json
  -rw-r--r--        70  27-Oct-2020  08:59:56  index.jelly
  drwxr-xr-x         0  27-Oct-2020  08:59:58  META-INF/maven/
  drwxr-xr-x         0  27-Oct-2020  08:59:58  META-INF/maven/io.jenkins.blueocean/
  drwxr-xr-x         0  27-Oct-2020  08:59:58  META-INF/maven/io.jenkins.blueocean/blueocean-executor-info/
  -rw-r--r--      4010  27-Oct-2020  08:59:56  META-INF/maven/io.jenkins.blueocean/blueocean-executor-info/pom.xml
  -rw-r--r--       130  27-Oct-2020  08:59:58  META-INF/maven/io.jenkins.blueocean/blueocean-executor-info/pom.properties
- ----------  --------  -----------  --------  --------------------------------------------------------------------------
                  6448                         10 files
</pre>
</div></div>

<p>If you're seeing high cpu usage with blue ocean 1.24.2 installed, it is not a result of the blueocean-executor-info plugin described in this bug report.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=paulsbruce" class="user-hover" rel="paulsbruce">paulsbruce</a> if you've upgraded to blue ocean 1.24.2 and restarted your Jenkins server since the upgrade, then the blueocean-executor-info plugin described in this bug report cannot be causing the high CPU usage.  With release 1.24.2, the blueocean-executor-info plugin contains no code.</p>

<p>The contents of the plugin file (hpi / jpi file) are:</p>



```
M Filemode      Length  Date Time      File
- ----------  --------  -----------  --------  --------------------------------------------------------------------------
  drwxr-xr-x 0  27-Oct-2020  08:59:58  META-INF/
  -rw-r--r--       855  27-Oct-2020  08:59:56  META-INF/MANIFEST.MF
  drwxr-xr-x 0  27-Oct-2020  08:59:58  WEB-INF/
  drwxr-xr-x 0  27-Oct-2020  08:59:58  WEB-INF/lib/
  -rw-r--r--       531  27-Oct-2020  08:59:58  WEB-INF/licenses.xml
  -rw-r--r--      3648  27-Oct-2020  08:59:58  WEB-INF/lib/blueocean-executor-info.jar
  drwxr-xr-x 0  27-Oct-2020  08:59:58  META-INF/maven/
  drwxr-xr-x 0  27-Oct-2020  08:59:58  META-INF/maven/io.jenkins.blueocean/
  drwxr-xr-x 0  27-Oct-2020  08:59:58  META-INF/maven/io.jenkins.blueocean/blueocean-executor-info/
  -rw-r--r--      4010  27-Oct-2020  08:59:56  META-INF/maven/io.jenkins.blueocean/blueocean-executor-info/pom.xml
  -rw-r--r--       130  27-Oct-2020  08:59:58  META-INF/maven/io.jenkins.blueocean/blueocean-executor-info/pom.properties
- ----------  --------  -----------  --------  --------------------------------------------------------------------------
  9174 11 files

```

<p>The contents of the jar file packaged in that plugin are:</p>



```
M Filemode      Length  Date Time      File
- ----------  --------  -----------  --------  --------------------------------------------------------------------------
  drwxr-xr-x 0  27-Oct-2020  08:59:58  META-INF/
  -rw-r--r--       855  27-Oct-2020  08:59:56  META-INF/MANIFEST.MF
  -rw-r--r--93  27-Oct-2020  08:59:56  jenkins-js-extension.json
  -rw-r--r--      1290  27-Oct-2020  08:36:28  package.json
  -rw-r--r--70  27-Oct-2020  08:59:56  index.jelly
  drwxr-xr-x 0  27-Oct-2020  08:59:58  META-INF/maven/
  drwxr-xr-x 0  27-Oct-2020  08:59:58  META-INF/maven/io.jenkins.blueocean/
  drwxr-xr-x 0  27-Oct-2020  08:59:58  META-INF/maven/io.jenkins.blueocean/blueocean-executor-info/
  -rw-r--r--      4010  27-Oct-2020  08:59:56  META-INF/maven/io.jenkins.blueocean/blueocean-executor-info/pom.xml
  -rw-r--r--       130  27-Oct-2020  08:59:58  META-INF/maven/io.jenkins.blueocean/blueocean-executor-info/pom.properties
- ----------  --------  -----------  --------  --------------------------------------------------------------------------
  6448 10 files

```

<p>If you're seeing high cpu usage with blue ocean 1.24.2 installed, it is not a result of the blueocean-executor-info plugin described in this bug report.</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `relates_to` from Jira: <a href="https://github.com/jenkinsci/blueocean-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-56833%22">JENKINS-56833</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `relates_to` from Jira: <a href="https://github.com/jenkinsci/blueocean-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-56833%22">JENKINS-56833</a>]</i>
